guides/requests/examples/1-auth.md

```diff
@@ -0,0 +1,167 @@
+
+Many public APIs require authentication. Most common pattern nowadays is the use of an `Authorization` header with a bearer token.
+
+```HTTP
+GET /api/companies?fields[company]=name&fields[employee]=name,profileImage&included=ceo&page[size]=10 HTTP/2
+Accept: application/vnd.api+json; profile="https://jsonapi.org/profiles/ethanresnick/cursor-pagination"
+Authorization: Bearer <token>
+```
+
+### Basic example
+
+In Ember Data we can create our own custom handler to add authentication header to all requests
+
+```js
+const ourSecureToken = '<token>'
+const AuthHandler = {
+ request({ request }, next) {
+    const headers = new Headers(request.headers);
+    headers.append(
+      'Authorization',
+      `Bearer ${ourSecureToken}`,
+    );
+
+    return next(Object.assign({}, request, { headers }));
+  }
+}
+```
+
+This handler would need to be added to request manager service configuration:
+
+```js
+export default class extends RequestManager {
+  constructor(args) {
+    super(args);
+
+    this.use([AuthHandler, Fetch]);
+    this.useCache(CacheHandler);
+  }
+}
+```
+
+This way every request that was made using this request manager will have `Authorization` header added to it.
+
+### Class based Handler
+
+Handlers can also be defined as classes. This is useful when you need to inject some services into your handler.
+
+Lets imagine we are using [Ember Simple Auth](https://github.com/simplabs/ember-simple-auth) addon to handle authentication. In this case we need to inject `session` service to get the token.
+
+**app/services/auth-handler.js**
+
+```js
+import { inject as service } from '@ember/service';
+
+export default class AuthHandler {
+  @service session;
+
+  request({ request }, next) {
+    const headers = new Headers(request.headers);
+    headers.append(
+      'Authorization',
+      `Bearer ${this.session.accessToken}`,
+    );
+
+    return next(Object.assign({}, request, { headers }));
+  }
+}
+```
+
+> **Note**
+> This is just a native javascript class, so it is not aware of Ember's dependency injection system.
+
+To use this handler we need to register it in our request manager service, but also we need to tell Ember's dependency injection system to provide context for `@service session;` to work.
+
+**app/services/request-manager.js**
+
+```js
+import RequestManager from '@ember-data/request';
+import Fetch from '@ember-data/request/fetch';
+import { CacheHandler } from '@ember-data/store';
+import { getOwner, setOwner } from '@ember/application';
+import AuthHandler from './auth-handler';
+
+export default class extends RequestManager {
+  constructor(args) {
+    super(args);
+
+    const authHandler = new AuthHandler();
+    setOwner(authHandler, getOwner(this));
+
+    this.use([authHandler, Fetch]);
+    this.useCache(CacheHandler);
+  }
+}
+```
+
+Apart of this little twist with dependency injection, everything else is the same as in previous example.
+
+## CSRF Token
```

Comment:
A little known thing regarding CSRF, is that as long as:

1. You have the client set a custom request header, 
2. check for it on the backend and
3. CORS isn't whitelisting all domains (`*`)

you don't need to fiddle with CSRF tokens.

I'm not saying we should change this section of the docs though, because some backends may still require this and maybe the backend is managed by another team.

But we could do something like this:

```suggestion
## CSRF Token

### Simple

The easy way of protecting against Cross Site Request Forgery (CSRF) is to set a static custom header in every HTTP Request (for example `X-CSRF-Protection: static`), check for it on the backend and make sure that CORS is only accepting data from trusted domains.

If you do this, you don't have to generate dynamic CSRF tokens for every request.

More information:

https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests

### Advanced

```

(then put the current text under advanced)

Reply:
Thanks for review and this proposal. I think it makes sense to add it. @runspired what do you think ? Should I also put code related to adding header in simple section ?